### ISSUE TITLE
Laptop wakes up from suspend on power change

### ISSUE THREAD

**Issue by Ganneff** [Dasharo/dasharo-issues#783]
### Component

Dasharo firmware

### Device

NovaCustom NS5x 11th Gen

### Dasharo version

1.5.2

### Brief summary

Suspend laptop. Plug in/out power -> it runs again

### How reproducible

100%

### How to reproduce

Suspend the laptop. Plug in (or out) power. Laptop active again. (Current Debian bookworm. Laptop booted in firmware update mode tells me there is no update available, so current stable version in use).

### Expected behavior

Stay in suspend.

### Actual behavior

Laptop active.

### Solutions you've tried

Nothing in Bios/Uefi settings to adjust this.
Tried adjusting files/values in /sys/class/wakeup. Doesnt seem to work.

**Comment by wessel-novacustom** [comment]
> Have you tried the `Power state after power failure` option in the UEFI setup menu @Ganneff ? It should be responsible for this behavior
> 
> On the other hand, on Heads we do not have this option, and by default this feature should be disabled. Not sure what's causing it to be enabled yet, but it seems like a separate issue

@mkopec This is not a Heads related issue, but UEFI instead. The issue @PeterLacknase is actually this: https://github.com/Dasharo/dasharo-issues/issues/766

@mkopec I think you might not have understood the issue correctly. It is not powering on from S0 state when plugging in, but it is powering on from **S0ix** state when plugging in. I think this is as expected for modern standby, but is it possible to change this behaviour either by firmware or software so that the charger can be connected when the laptop is in S0ix, without the screen display turning on?

**Comment by mkopec** [comment]
Ah, thanks for clarifying. I believe it should be possible to make it so that inserting a power supply does not wake the laptop.

**Comment by wessel-novacustom** [comment]
> Ah, thanks for clarifying. I believe it should be possible to make it so that inserting a power supply does not wake the laptop.

@mkopec How? By firmware or on OS-side?

**Comment by Ganneff** [comment]
> Have you tried the `Power state after power failure` option in the UEFI setup menu @Ganneff ? It should be responsible for this behavior

While Wessel already clarified its happening from Suspend to idle S0ix, still wanted to confirm, i set the option as <Powered Off>. Unfortunately, no other option around to make it stop this.

**Comment by mkopec** [comment]
@Ganneff @wessel-novacustom Could you clarify if this issue only happens on Debian? So far I haven't been able to reproduce on Ubuntu on NV4x and NS5x 11th gen

**Comment by PeterLacknase** [comment]
It's completely OS independent. To reproduce:

[1] Shut down NV41 (used over here).
[2] Plug power cable from power adapter into a/the wall socket.
[3] Connect power adaptor with the laptop (preferably lid closed).

-> The laptop powers on.

**Comment by mkopec** [comment]
@PeterLacknase As @wessel-novacustom mentioned above, your issue with Heads seems like https://github.com/Dasharo/dasharo-issues/issues/766 and it will be fixed in the next firmware relase

This issue is for waking from s0ix suspend (sleep mode) if I'm understanding this thread correctly

@wessel-novacustom @Ganneff Can you share more about the setup where this issue happens? I installed Debian 12 on NS5x 11th Gen and I cannot reproduce this issue. Are you testing with regular AC adapter or an USB-C adapter?

**Comment by PeterLacknase** [comment]
Right. That’s an issue, too. But it affected sleep state as well. In the mean time I found out, that it only happens with some USB-Sticks connected. With others (like nitrokey) everything works as expected.

**Comment by wessel-novacustom** [comment]
> @wessel-novacustom @Ganneff Can you share more about the setup where this issue happens? I installed Debian 12 on NS5x 11th Gen and I cannot reproduce this issue. Are you testing with regular AC adapter or an USB-C adapter?

- @Ganneff A video would be very helpful.

**Comment by Ganneff** [comment]
I have a normal Debian 12 setup. Disk encrypted, but standard Debian installer setup. UEFI Bios secure boot things not turned off.
KDE environment.

I do use USB-C for power.

Video: What do you want to see? Me using my phone showing how i unplug and it turns back on? A screenrecording of something?

**Comment by wessel-novacustom** [comment]
> Video: What do you want to see? Me using my phone showing how i unplug and it turns back on?

- Yes, exactly, so that the firmware developers can overview this behaviour and let you know if there is something we can do about it.

**Comment by philipandag** [comment]
I will try to reproduce this configuration as closely as possible
> I have a normal Debian 12 setup. Disk encrypted, but standard Debian installer setup. UEFI Bios secure boot things not turned off.
KDE environment.
>
>I do use USB-C for power.

**Comment by philipandag** [comment]
I have installed Debian 12.6 Bookworm stable with encryption and KDE. Secure Boot on. Used both USB-C and barrel jack adapters. Dasharo v1.5.2 release. No USB devices. 

I could not reproduce the issue of the laptop waking up from suspension when a power adapter is connected or disconnected during suspension.
Tried with both S3 and S0ix modes.

The laptop does power on if it is shut down and then the power adapter is connected. This behavior is controlled by Dasharo System Features -> Power Management Options -> Power state after power failure and setting it to "Powered off" causes the laptop to stay powered off after plugging in the power adapter.
